File: Python3/string-to-integer-atoi.py
import re

class Solution:
    def myAtoi(self, s: str) -> int:
        pattern = r'^\s*([-+]?\d+)'
        match = re.search(pattern, s)
        if match:
            return min(
                max(
                    int(match.group(1)),
                    -2147483648
                ),
                2147483647
            )
        else:
            return 0

File: Golang/string-to-integer-atoi.go
import (
	"strconv"
	"math"
	"regexp"
    "strings"
)

func myAtoi(s string) int {
    s = strings.TrimSpace(s)
    pattern := `^([-+]?\d+)`
    regex := *regexp.MustCompile(pattern)
    match := regex.Find([]byte(s))
    matchAtoi, _ := strconv.Atoi(string(match[:]))

    return min(max(matchAtoi, math.MinInt32), math.MaxInt32)
}
